tests/unit_tests/utils/test_names.py

```diff
@@ -170,102 +170,144 @@ def test_get_telescope_type_from_telescope_name():
             names.get_telescope_type_from_telescope_name(_name)
 
 
-def test_camera_efficiency_names():
-    """
-    Test the various camera efficiency names functions
-    """
+def test_generate_file_name_camera_efficiency():
 
     site = "South"
-    telescope_model_name = "LST-1"
+    telescope_model_name = "LSTS-01"
     zenith_angle = 20
     azimuth_angle = 180
     label = "test"
+
     assert (
-        names.camera_efficiency_results_file_name(
-            site, telescope_model_name, zenith_angle, azimuth_angle, label
+        names.generate_file_name(
+            "camera-efficiency-table",
+            ".ecsv",
+            site,
+            telescope_model_name,
+            zenith_angle,
+            azimuth_angle,
+            label=label,
         )
-        == "camera-efficiency-table-South-LST-1-za020deg_azm180deg_test.ecsv"
+        == "camera-efficiency-table-South-LSTS-01-za20.0deg_azm180deg_test.ecsv"
     )
+
     assert (
-        names.camera_efficiency_simtel_file_name(
-            site, telescope_model_name, zenith_angle, azimuth_angle, label
+        names.generate_file_name(
+            "camera-efficiency",
+            ".dat",
+            site,
+            telescope_model_name,
+            zenith_angle,
+            azimuth_angle,
+            label=label,
         )
-        == "camera-efficiency-South-LST-1-za020deg_azm180deg_test.dat"
+        == "camera-efficiency-South-LSTS-01-za20.0deg_azm180deg_test.dat"
     )
+
     assert (
-        names.camera_efficiency_log_file_name(
-            site, telescope_model_name, zenith_angle, azimuth_angle, label
+        names.generate_file_name(
+            "camera-efficiency",
+            ".log",
+            site,
+            telescope_model_name,
+            zenith_angle,
+            azimuth_angle,
+            label=label,
         )
-        == "camera-efficiency-South-LST-1-za020deg_azm180deg_test.log"
+        == "camera-efficiency-South-LSTS-01-za20.0deg_azm180deg_test.log"
     )
 
     site = "North"
-    telescope_model_name = "MST-FlashCam-D"
+    telescope_model_name = "MSTN"
     zenith_angle = 40
     azimuth_angle = 0
     label = "test"
     assert (
-        names.camera_efficiency_results_file_name(
-            site, telescope_model_name, zenith_angle, azimuth_angle, label
+        names.generate_file_name(
+            "camera-efficiency-table",
+            ".ecsv",
+            site,
+            telescope_model_name,
+            zenith_angle,
+            azimuth_angle,
+            label=label,
         )
-        == "camera-efficiency-table-North-MST-FlashCam-D-za040deg_azm000deg_test.ecsv"
+        == "camera-efficiency-table-North-MSTN-za40.0deg_azm000deg_test.ecsv"
```

Comment:
Related to previous comment, also seen later a couple of times.

Reply:
Same reply - I've commented on zenith vs azimuth accuracy. Let me know if this makes sense, easy to change it to `azm000.0deg`, but I don't think it makes sense (and python is horrible in getting the accuracy right)